UI/UX enhancements while setting up accounts

eswarasai opened the pull request:
### Description
 - [x] Added temporary password to be saved for 5 minutes in the background
 - [x] Added toggle to show/hide password input field
 - [x] Added expiration for mnemonic regeneration
 - [x] Add expiration time left status on UI

Fixes: #84

eswarasai commented:
@ttaoS @anjj123 - This is ready for review except for the piece of displaying the time left status on the UI. Let me know if there's any other change needed alongside this. Thanks!

ttaoS commented on src/ts/background/services/keyring-vault.ts:
```diff
@@ -24,6 +25,23 @@ class KeyringVault {
+  clearTempPassword (): void {
+    setTimeout(() => {
+      this._tempPassword = undefined
+    }, 300 * 1000) // 5 minutes
```
we probably should keep it in a config file

eswarasai replied:
Sure, I'll do that 👍

ttaoS commented on src/ts/background/services/keyring-vault.ts:
```diff
@@ -80,10 +98,23 @@ class KeyringVault {
+      // Clear interval for _mnemonic
+      setTimeout(() => {
+        this.clearMnemonic()
+      }, 300 * 1000) // 5 minutes
```
same above

ttaoS commented:
It looks good overall. Are you still working on `Add expiration time left status on UI`?

eswarasai commented:
@ttaoS -- I haven't started working on displaying the expiration time left yet. Let me know if there's a suggested way to implement the timer apart from running the `setTimeout` in the background every second. Thanks!

Web3Foundation commented:
@ttaos @mytechtip is this one good to pay out now?

ttaoS commented on src/ts/background/store/wallet/reducer.ts:
```diff
@@ -4,15 +4,18 @@ import { ACTION_TYPES, IAccount } from './actions'
   locked: true,
-  created: false
+  created: false,
+  expiryTimeLeft: 0
```
I would call it more specific e.g`tempPwdExpiryTimeLeft`

eswarasai replied:
It can either be temp password or the mnemonic. If you'd like it to be more specific, then we'd need to create 2 separate variables to keep track of which seems unnecessary to me. Does that make sense?

ttaoS replied:
right now, if i look at it, i would guess it is exprit time for wallet.

eswarasai replied:
True. But all the account setup related variables are within `Wallet` state. Hence I've added `expiryTimeLeft` to its state.

I can add a comment beside `expiryTimeLeft` mentioning that it's while setting up an account. If you'd like me to refactor the variable name across the codebase to something else like `expiryTimeLeftForAccountSetup`, I can do that as well.

ttaoS replied:
Is it the same expiry time left for both password and mnemonic?

eswarasai replied:
Yep. Because the password timer expiry will only happen once at the time of setting up the wallet for first time during which it's valid for mnemonic as well.

Later, when always setting up a new account, the `expiryTimerLeft` will only be applicable to mnemonic since the wallet has been setup already.

ttaoS replied:
I see. In this case, we probably can call it `expiryTimeLeftForAccountSetup` as you suggested.

eswarasai replied:
Should I do the above refactor throughout the codebase including the methods like `getExpiryTimeLeft` as well as in the `keyring-vault.ts` like `_expiryTimeLeftTimerId`?

Honestly, `expiryTimeLeftForAccountSetup` is a pretty long variable name and I'm more biased towards having a comment that explains its usage rather than refactoring across. I'll leave it upto you though.

ttaoS replied:
@anjj123 @fyang1024 @mytechtip any thought on this?

eswarasai replied:
Any update on this? Thanks!

mytechtip replied:
How about calling it `timeoutForAccountSetup` or `accountSetupTimeout`?

eswarasai replied:
@mytechtip - Sure, I'll update it to `accountSetupTimeout `. Although, I'd still keep the `expiryTimeLeft` in the `keyring-vault.ts` and add a little comment beside mentioning that this timer would be for password/mnemonic expiry. Does that make sense?

ttaoS commented on src/ts/background/services/keyring-vault.ts:
```diff
@@ -23,6 +27,47 @@ class KeyringVault {
+  }
+
+  clearTempPassword (): void {
+    this.startExpiryTimer()
```
why do we need to start expiry timer inside clear temp password func?

eswarasai replied:
Ahh. We don't. I can move it up to `setTempPassword()` real quick. Thanks for pointing it out.

ttaoS commented on src/ts/background/services/keyring-vault.ts:
```diff
@@ -6,15 +6,19 @@ import { mnemonicGenerate, cryptoWaitReady, mnemonicValidate } from '@polkadot/u
   private _mnemonic?: string
   private simpleAccounts?: SimpleAccounts
+  private _expiryTimeLeft = 0
+  private _expiryTimeLeftId = 0
```
_expiryTimeLeftId -> _expiryTimeLeftTimer?

eswarasai replied:
I think `_expiryTimeLeftTimerId` makes more sense since it's the `timerId` that is being returned by the `setInterval` function. Let me know what you think. Thanks!

ttaoS replied:
yup.

ttaoS commented on src/ts/background/services/keyring-vault.ts:
```diff
@@ -23,6 +27,48 @@ class KeyringVault {
+  startExpiryTimer (): void {
+    this._expiryTimeLeft = VALIDITY_INTERVAL
+
+    this._expiryTimeLeftTimerId = window.setInterval(() => {
```
I wonder if we can use Timeout instead interval here. The benefit of it is that popup client doesn't need call `getExpiryTimeLeft` to clear it.

eswarasai replied:
The reason behind using `setInterval` is to update the time left for expiry of either password/mnemonic to display it in the UI. If we use `setTimeout`, it executes only once and not each and every second to update the time left on the UI.

ttaoS replied:
i am not referring to clear pwd or mnemonic part. I am talking about the expiry time left timer bit. But i think i get what you mean, it is ok then. 
![image](https://user-images.githubusercontent.com/37568337/63774225-2184d280-c920-11e9-978f-46207e4c7f52.png)

eswarasai replied:
Ahh. My bad. For some weird reason, I've observed that the `clearInterval` wasn't being triggered and the `expiryTimeLeft` was going negative. Hence I've added the above check explicitly.

Web3Foundation commented:
@eswarasai @ttaoS what's the latest with this PR?

ttaoS commented:
> @eswarasai @ttaoS what's the latest with this PR?

It is ok on my end once the field name is changed to `accountSetupTimeout` as @mytechtip suggested. I think @anjj123 may want to review UI change before we merge it

anjj123 commented:
@eswarasai @ttaoS I found quite a few issues (i was testing on Firefox)

General Issues:
- Timer should be in a consistent place on the screen (I recommend just having the countdown in a slightly transparent font in the top just below the 1-2-3 icons)
- Timer doesn't show when first creating an account, unless you close and reopen the extension
- Timer doesn’t seem to be tracking consistently on creation (when click off and on the extension, depending on what screen you’re on, the timer tends to decrease or increase or disappear completely)

Initial Account Creation:
- Timer should also show on the confirm mnemonic screen
- When timer elapses, should take user back to the very first screen where they pick their colour

Creating a second account:
- Timer ends but it still lets you create the account while the extension is still open (should return to home screen)
- Endlessly forced to keep creating accounts unless you close and open the extension
- Needs a quit button; otherwise, user is forced to sit there waiting out the timer (we can do this one ourselves - i need to organise the design for this - just noted here as a reminder)

eswarasai commented:
@anjj123 - Thanks for letting me know. I'll take a look at the above issues and put up a fix soon.

eswarasai commented:
@anjj123 - Sorry about the delay in updating this PR. I've fixed almost all the above mentioned UI/UX issues. Thanks for pointing them out and hopefully there aren't any issues anymore in regards with the expiry timer. In case you find any, please do let me know along with the reproduction steps so that I can fix them immediately.

In continuation with my above comment, I need some help in terms of Design to wrap up the below 3 tasks:
- Timer should be in a consistent place on the screen (I recommend just having the countdown in the top right just below or above the 1-2-3 icons with a rounded shaded box around thats the same colour as the theme selected and the text in white - needs to stand out)
- Timer should also show on the confirm mnemonic screen
- Needs a quit button; otherwise, user is forced to sit there waiting out the timer (we can do this one ourselves - i need to organise the design for this - just noted here as a reminder)

Let me know how you'd suggest to move forward and wrap up this one. Thanks!

cc: @ttaoS

fyang1024 commented:
Hi @eswarasai , we have done a quick test and found issues which can be reproduced as below:

1. Enter password twice and click create new account
2. See mnemonic screen
3. Close browser extension
4. Open browser extension. **We expect to see mnemonic screen**, but instead we see password screen

1.  Enter password twice and click create new account
2. See mnemonic screen
3. Go to confirm mnemonic screen
4. Close browser extension
5. Open browser extension. **We expect to see confirm mnemonic screen**, but instead we see password screen

eswarasai commented:
@fyang1024 - That's the expected behaviour. Please refer to this comment for better explanantion -- https://github.com/GetSpeckle/speckle-browser-extension/issues/84#issuecomment-519411487

fyang1024 commented:
> @fyang1024 - That's the expected behaviour. Please refer to this comment for better explanantion -- [#84 (comment)](https://github.com/GetSpeckle/speckle-browser-extension/issues/84#issuecomment-519411487)

I think it's misunderstanding, if you look at the original issue https://github.com/GetSpeckle/speckle-browser-extension/issues/84.
What @ttaoS means is unmask feature on the password screen, not always go back to password screen (that's the original flow we want to change).
Hope it clarifies.

ttaoS commented:
> > @fyang1024 - That's the expected behaviour. Please refer to this comment for better explanantion -- [#84 (comment)](https://github.com/GetSpeckle/speckle-browser-extension/issues/84#issuecomment-519411487)
> 
> I think it's misunderstanding, if you look at the original issue #84.
> What @ttaoS means is unmask feature on the password screen, not always go back to password screen (that's the original flow we want to change).
> Hope it clarifies.

Yeah. Fei is right. User can go back to password step and unmask if they want to.

eswarasai commented:
@fyang1024 @ttaoS - Thanks for the clarification. I'll look into this behaviour now.

eswarasai commented:
@ttaoS -- Is it fine, if I keep track of the `lastVisitedRoute` in the `keyring-vault.ts` via the messaging method similar to that of `tempPassword` and `mnemonic`? Not sure what's the best way to implement the above desired behaviour. Let me know if you have any better ideas. Thanks!

cc: @fyang1024 @anjj123

ttaoS commented:
> @ttaoS -- Is it fine, if I keep track of the `lastVisitedRoute` in the `keyring-vault.ts` via the messaging method similar to that of `tempPassword` and `mnemonic`? Not sure what's the best way to implement the above desired behaviour. Let me know if you have any better ideas. Thanks!
> 
> cc: @fyang1024 @anjj123

Or you can use temp password and mnemonic to determine the route. Eg if both are populated then go to mnemonic otherwise go to password step.

eswarasai commented:
@ttaoS - I thought of that approach but it's a little less reliable because of even if I have generated the mnemonic and moved back to password step and close the app, next I'll be taking the user to confirm mnemonic screen and not the password step. If this behaviour is fine, I can follow your suggested approach. For better reliability, I can implement the way I've suggested above. Let me know which one should I go ahead with. Thanks!

ttaoS commented:
> @ttaoS - I thought of that approach but it's a little less reliable because, even if I have generated the mnemonic and moved back to password step and close the app, next time I'll be taking the user to confirm mnemonic screen and not the password step. If this behaviour is fine, I can follow your suggested approach. For better reliability, I can implement the way I've suggested above. Let me know which one should I go ahead with. Thanks!

I feel it is ok as we just want to provide convenience to users so that they can set up accounts quickly. Not a bookmark where  they visited last.

eswarasai commented:
@ttaoS - Also, one more thing, if I have to follow the logic of checking for `tempPassword` and `mnemonic`, user will always be viewing the `Confirm Phrase` UI upon opening the popup, until the timer expires, if the password is already set and never the `Generate Phrase` UI as the mnemonic would've already been set when this UI is being displayed. I just wanted to update you on this behaviour. Same applied for the case of `Creating New Account` UI

ttaoS commented:
> @ttaoS - Also, one more thing, if I have to follow the logic of checking for `tempPassword` and `mnemonic`, user will always be viewing the `Confirm Phrase` UI upon opening the popup, until the timer expires, if the password is already set and never the `Generate Phrase` UI as the mnemonic would've already been set when this UI is being displayed. I just wanted to update you on this behaviour. Same applied for the case of `Creating New Account` UI

I thought the temp mnemonic is only generated when you click confirm. If it is the case, I think it should be ok. Btw, you may want to add a cancel button so that users can skip account setup before time expires for creating a new account process. Not sure if we should add some text in timer as it may confuse ppl what it is for @anjj123 any thought?

eswarasai commented:
@ttaoS - I need some help w.r.t. some UI decisions in regards with the timer especially. Please find them here in one of the above [comment](https://github.com/GetSpeckle/speckle-browser-extension/pull/108#issuecomment-531522504)

eswarasai commented:
@ttaoS @fyang1024 - I've just pushed my changes on handling the step to be viewed upon re-open of popup. Please note that based on the above discussed logic, if user starts creating account by setting the password, he/she will always be taken to Confirm Phrase UI and not Generate Phrase UI or Set Password UI upon popup re-open.

Let me know if there's anything else needed. Thanks!

PS: I had to add `tempAccountName` to the `keyring-vault` to keep track of the account name user is setting in the Generate Phrase UI, which wasn't handled earlier.

anjj123 commented:
@eswarasai I provide UI screens below for the design of the timer and the two buttons. 

Timer sits here once the user generates a mnemonic. Persists for all relevant screens (including confirm mnemonic). Color is contingent on the color picked by user at the start:
![Screen Shot 2019-09-06 at 4 18 18 pm](https://user-images.githubusercontent.com/39804936/64929493-3a6f0c80-d86a-11e9-9159-a9d85788dc6b.png)

Generate menmonic now has these two buttons. If cancelled, the timer is cancelled and the user will generate a new seed after entering their password. Cancel sends user back to PW screen:
![Screen Shot 2019-09-06 at 4 22 29 pm](https://user-images.githubusercontent.com/39804936/64929507-52df2700-d86a-11e9-8360-655289d634ed.png)

Confirm mnemonic screen:
![Screen Shot 2019-09-06 at 4 21 24 pm](https://user-images.githubusercontent.com/39804936/64929521-74401300-d86a-11e9-9aff-bb35b0c58cfe.png)

eswarasai commented:
@anjj123 - The designs look great, although a bit tricky to implement the timer on the Header. I'll take a look into it and update the PR accordingly.

Let me know in the meantime if there's any other functional issues with the latest code. Thanks!

anjj123 commented:
@eswarasai it doesnt need to be there per se. It just has to be on a consistent spot on each screen, so its not jumping around between screens. if you think there's an easier/better spot to place it, feel free to experiment :)

eswarasai commented:
@anjj123 - Honestly, I couldn't come up with a better spot either. Confirm Phrase screen is a bit cluttered on the content section. Hence having it on the Header above the Steps flow is the only spot left to utilize. I'll play around with it, no worries 🙂

eswarasai commented:
@anjj123 - I've updated the design as well as incorporated the cancel account setup functionality like you're suggested above. Let me know if there's anything else pending.

@ttaoS @fyang1024 - Can you guys also please do one final review?

Sorry about the long delay on this one 😅

Web3Foundation commented:
@ttaoS @fyang1024 @eswarasai what's the latest here?

eswarasai commented:
@Web3Foundation - I'm awaiting on the final review for this PR. Hopefully this'll be merged in the next couple of days, if there are no further changes needed 🙂 

cc: @ttaoS @anjj123 @fyang1024

eswarasai commented:
@ttaoS @anjj123 @fyang1024 - Umm, any update here? Would really like to close this one after spending so much time and efforts on this 😅

anjj123 commented:
@eswarasai hey sorry for the delay, just got back from China. Will check over the weekend. Thanks for this!

anjj123 commented:
@eswarasai Looks great!

Bugs:
1. Cosmetic issues:
[Chrome] If a colour other than blue is picked, the 'Accept Terms of Service' screen doesn't change the colour of the top banner (the dots remain blue) 
![Screen Shot 2019-10-09 at 2 52 53 am](https://user-images.githubusercontent.com/39804936/66412535-72b8d380-ea41-11e9-893c-df2b2c3f9b20.png)


[Firefox] The glow around the dots and the box for the timer remain blue
![Screen Shot 2019-10-09 at 2 54 02 am](https://user-images.githubusercontent.com/39804936/66412521-6af92f00-ea41-11e9-9d49-343e3b78c778.png)

2. If i am on the seed phrase screen and minimise the extension, the extension returns me to the confirm seed phrase screen - it should return the user to whichever screen they left off;

3. [Firefox] The timer elapses when there is still around ~6 seconds remaining.

4. The Account Names no longer save now and instead default to "N/A" once created

5. When creating a second account, the screen will randomly close back to the home screen before timer elapses (unknown cause)

Once these are fixed, we will do one last code review and flag it for approval.

eswarasai commented:
@anjj123 - Please find my comments below:

1.a) I've fixed the issue with Header component not updating upon receiving props as it was a functional component. This issue is unrelated to my changes.

1.b) I've updated the header shadow to be that of color scheme for Steps. As per the background color for the timer, it's using the shadow color property of the color scheme. Let me know if there's any change needed here. Also, these are again unrelated changes for this PR.

2) This has already been discussed. Please refer to the above [comment](https://github.com/GetSpeckle/speckle-browser-extension/pull/108#issuecomment-531604396)

3), 4) and 5) I'm unable to reproduce this issue. Can you please guide me with the steps? Also, please ensure you have the latest changes as I've made sure that Account name is being saved appropriately in the previous change.

Thanks!

anjj123 commented:
Thanks for the quick response @eswarasai 

1. Heading colour issue looks fixed for every colour but purple. 

2. I see Tony mentioned it above. If it's not too much trouble, it would be great to have it return the user to the screen they left off on. I believe his comment was related to returning the user to the PW screen. If the user hasnt proceeded to mnemonic confirmation, I think it's fine to leave them on the generated seed so they can write it down somewhere. 

3 + 5. I'll do a screen recording later tonight but it seems to be quite inconsistent. The steps I've generally taken are as follows:
i. Fresh install of extension
ii. Create first account
iii. Proceed to create a second account off the home screen
iv. After waiting some time for the timer to elapse, it randomly closes after a minute or two (at times immediately)
I haven't been able to pin it down to a root cause yet.

4. I'm still getting account names defaulting to N/A if I don't manually enter an account name - is this the same on your end if you leave the pre-generated Account Names ("Account 1, Account 2" and so on)?

eswarasai commented:
@anjj123 - Thanks for your quick response.

1) I've double checked all the colors and it did work fine at that moment. Let me double check and see if I've missed something in regards to this inconsistent behaviour.

2) For me to implement this desired behaviour, I need to keep track of which step user has closed the popup from. @ttaoS had responded to this exact comment and was fine with not taking user back to generate seed step even though he/she hasn't moved to confirm mnemonic yet.

3) Sure, thanks for the steps. I'll also try to reproduce the behaviour on my end.

4) I missed the pre-generated Account names handling. Thanks for pointing it out 👍

eswarasai commented:
@anjj123 - I've fixed below issues:

1. Heading colour issue, please do double-check.

2. The timer elapses intermittently. This was caused because of adding the cancel feature where I wasn't clearing the `setTimeout` timer. So if you cancel and again try creating a new account, the previous timer was getting executed.

For the account names defaulting to `N/A`, I wasn't able to reproduce this. You can find the relevant screenshots below:

<img width="376" alt="Screenshot 2019-10-09 at 6 58 55 PM" src="https://user-images.githubusercontent.com/5172086/66490345-a130c000-eace-11e9-82bf-4ee13e8a7fee.png">

<img width="380" alt="Screenshot 2019-10-09 at 6 59 58 PM" src="https://user-images.githubusercontent.com/5172086/66490356-a4c44700-eace-11e9-948b-8e353f784741.png">

anjj123 commented:
@eswarasai 

Thanks for this!

1.  I'm having the N/A issue still on account names (even when manually entering a name). I made a recording
![bug](https://user-images.githubusercontent.com/39804936/66897744-59b4b180-f043-11e9-9c4b-7199abfbbd40.gif)

2. When creating a new account, the page still randomly closes to the home screen (sometimes under 1 minute it will occur). Sometimes when closing and opening the extension again, the confirm mnemonic screen will reappear. Same steps as before. I can't pin it down to a consistent issue. This happens on both Firefox and Chrome.

3. The colour issue still exists for purple (the background glow is still blue and the box on the timer)
![Screen Shot 2019-10-16 at 6 21 55 pm](https://user-images.githubusercontent.com/39804936/66897612-12c6bc00-f043-11e9-8473-94893cf19f44.png)

eswarasai commented:
@anjj123 - I'm unable to reproduce both the issues 1) & 2) mentioned above. My guess is that you haven't pulled the latest changes. Can you please either do a clean installation of my branch on your system or rebase my branch by executing `git pull -r` and try again? I'm pretty confident on the code front everything is working fine as I've debugged the logs for all the scenarios while creating accounts.

As to answer your point 3), I'm using `shadowColor` of the theme for the `Timer` which you can find [here](https://github.com/GetSpeckle/speckle-browser-extension/pull/108/files#diff-f3574742f3d3167610a758906a68ca12R125). You might want to check the existing color codes of the theme for the purple and can let me know if you'd need me to update them. They're definitely not Blue as you can see the difference clearly in case of selecting Blue theme based on below screenshot.

<img width="377" alt="blue" src="https://user-images.githubusercontent.com/5172086/66935757-334e4080-f05a-11e9-952f-0a4512d6cdef.png">

<img width="376" alt="purple" src="https://user-images.githubusercontent.com/5172086/66935858-655fa280-f05a-11e9-9125-750347d40839.png">

Web3Foundation commented:
@anjj123 @eswarasai will complete once things are finalized here; this is a long winded issue! 🔢

anjj123 commented:
@eswarasai You're right about the colour. We can leave it. I was using the latest changes but I'll do some testing again tonight to confirm.

eswarasai commented:
@anjj123 - Any update here? Thanks!

anjj123 commented:
@eswarasai Hey, I've tested this a few times now and I'm still getting the same issues:

1. Accounts not being saved with their names ("N/A")
2. Mnemonic screen closing before timer elapses on second account creation.

I'm going to have @hskang9 do some tests to see if he's getting the same issue on his end.

eswarasai commented:
@anjj123 - Okay, thanks for trying it out again. I'll also try and double-check again on my end first thing tomorrow morning.

I'd really appreciate some thorough testing on this one from all of you guys: @ttaoS, @fyang1024 and @hskang9. Thanks 🙂

hskang9 commented:
https://www.loom.com/share/73a68d4b85ab48cfa76e8e34549f4f3d

So as I remember, every UI components were working well until I found this....

The problem in the video happened when rendering qrcode component with semantic UI. It seems like the qrcode component in semantic UI sets z-index position as absolute, but the rest of the other react component in the extension sets z-index position as relative. I was trying to set other component's position as absolute, then it broke all the layouts in flexbox. Due to this, I try not to use semantic UI component when there is overlapping UI in the extension. In conclusion, if you could render qrcode component with css in styled-component with position: relative, everything would be perfect on qr I believe.

eswarasai commented:
@hskang9 - Thanks for the quick test. The QR code issue doesn't fall under this PR but I hear you. Can you please create a new issue for the above bug?  I'd be more than happy to take a shot at it.

@anjj123 - Let me know how should we proceed here 😅

hskang9 commented:
> @anjj123 - Okay, thanks for trying it out again. I'll also try and double-check again on my end, first thing tomorrow morning.
> 
> I'd really appreciate some thorough testing on this one from all of you guys: @ttaoS, @fyang1024 and @hskang9. Thanks

Oh no my mistake, but it had to be discussed anyways. I am testing your PR as well. I will show you video soon.

hskang9 commented:
https://www.loom.com/share/6664c336fe4744a094b120811a0fcd9b

Hello @eswarasai, I found that all 4 items has passed, but as @anjj123 said, all accounts' name were saved as 'N/A'. The timer worked fine for me though. For the PR, it looks good to me, but if this breaks the current feature, I am not clear if I could merge this. However, I am trying to fix the issues to get your PR ready.

eswarasai commented:
@hskang9 - Thanks for taking a look into it and sharing the recording. I'm gonna debug further to reproduce the `N/A` issue and post an update here soon.

hskang9 commented:
No problem. Thank you for your contribution. As I test more, I found out that once the mnemonic regenrates, the account name shows 'N/A'. Let's debug this more 👷

eswarasai commented:
@hskang9 - Is it possible for us to jump on a call to debug this further? Unfortunately I'm still unable to reproduce this on my end and can't quite seem to figure out the exact root cause for this behaviour. I'm hoping it might be an environment related issue as I'm using MacOS and I can see from the above screenshare that you're on Ubuntu. Thanks 🙂

hskang9 commented:
Sure @eswarasai, we can test on multiple devices. I sent you the email. Please answer it.

eswarasai commented:
@hskang9 - Thanks for connecting with me and helping me test in trying to reproduce the issue. I'm glad that everything was working as expected on your machine. I've also pushed the color scheme update for Orange. Let me know if there's anything else needed here so that I can quickly try to resolve it.

eswarasai commented:
@hskang9 - Any update on this one? I see it hasn't been approved and merged yet. Let me know if there's anything needed from my end. Thanks!

hskang9 commented:
@eswarasai I found out how to reproduce the N/A account error. 
https://www.loom.com/share/e1216651b9fa44789921ded20f431cba

It looks like the problem is happening on the mnemonic verification UI. when I close the extension, the extension just drops the account name. I left the video for the explanation.

eswarasai commented:
@hskang9 - Thanks for the reproduction. I'll try to handle this. The problem is, I'm only saving the account name temporarily in background if user clicks on Create button and moves to the confirm phrase step. I'll try to update the account name when user closes the extension on generate phrase page.

eswarasai commented:
@hskang9 - Just pushed the fix.

I wasn't able to leverage the close event of the extension. Hence I'm updating the background continuously as the user inputs the account name, except for the case of no input for account name to avoid it being empty. We weren't handling that scenario so far 🤷‍♂

eswarasai commented:
@hskang9 - Any update here or any other issues pending that I need to look into? Thanks!

cc: @anjj123 @ttaoS

hskang9 commented:
@eswarasai now the current UI shows N/A every time in my computer. Did you fix the error on master branch?

hskang9 commented:
Hey @eswarasai I saw that you added temp account in the background with console.log() functions. However, the log is not printed and the accout name is always generated with 'N/A'.  However, once I input other characters than the default account, it works well.

eswarasai commented:
@hskang9 - I think I know the issue. Let me push a fix for this soon 😥

hskang9 commented:
It's okay @eswarasai. We all make mistakes. We appreciate that you are contributing to making this feature that we are longing for. Please give me an update when you catch the bug💪

eswarasai commented:
@hskang9 - I'm hoping that my commit fixes the `N/A` issue once and for all 🤞

hskang9 commented:
@eswarasai Unfortunately, the error is not fixed yet. The log seems like it drops the name after validating mnemonic.

hskang9 commented:
In GeneratePhrase.tsx, when I click the create button, `handleClick` only background's `setNewPhrase(mnemonic)`. Can you give a logic to put the account name in the background there so that when the UI is redirected to ConfirmPhrase.tsx, the component can get the account information in the background?

hskang9 commented:
If you need help, please do not hesitate to have a meeting.

hskang9 commented:
@eswarasai I think doing that fixes most of the issue, but I need to find out if the time passes, the account is removed in the background.

eswarasai commented:
@hskang9 - Please find my comments below:

> The log seems like it drops the name after validating mnemonic.
Yes, because the account is created with the given name, so we remove the temp account name

> Can you give a logic to put the account name in the background
This is the change I've done and it happens as soon as the GeneratePhrase page loads or when you change the account name in the background

Let's jump on a call to see the reproduction steps. Please send out a calendar invite for the same. Thanks!

hskang9 commented:
@eswarasai I do not see the changes we have gone through in the call. Also, do we need ‘setTempAccountName’ function?

eswarasai commented:
@hskang9 - I've pushed the change actually here --> https://github.com/GetSpeckle/speckle-browser-extension/pull/108/commits/811874aa3751af98656939136ca69f11c96ade3d

That was the only missing piece. Add yes, `setTempAccountName()` sets the account name temporarily in the background, so that we can retrieve it, even if the extension popup is closed and re-opened.

hskang9 commented:
LGTM

eswarasai commented:
@ttaoS @anjj123 @hskang9 - Can we please merge this? Thanks :)

fyang1024 commented:
@ttaoS @mytechtip @hskang9 how can we merge this to both release and master branch?
We probably don't want to merge to master first and then merge master to release, as master has some changes we don't want for release. We don't want opposite as well, since release has some changes we don't want for master.

mytechtip commented:
Might need to cherry pick

ttaoS commented:
> @ttaoS @mytechtip @hskang9 how can we merge this to both release and master branch?
> We probably don't want to merge to master first and then merge master to release, as master has some changes we don't want for release. We don't want opposite as well, since release has some changes we don't want for master.



> Might need to cherry pick

@fyang1024 yeah. we may have to cherry pick it if we want to release this feature.

hskang9 commented:
Hey @eswarasai so in UX perspective, we have the timer closed before it elapses the time. Can we make changing view logic `this.props.history.push` a callback for the timer so that view change is occurred after the timer ends?

hskang9 commented:
Ok @eswarasai the other thing that I found out is that the timer really closes in the middle of decreasing when I locked my computer off and on again. Can we store `timeleft` variable when the extension is closed?

hskang9 commented:
@eswarasai Kusama transfer is now lit 🔥🔥🔥🔥 I have reviewed the code for you to put the changes so that we can launch our extension.

hskang9 commented:
@eswarasai fire is burning 🔥 🔥🔥🔥🔥🔥👀👀👀👀👀
